src/server/rpc/jsdoc-extractor.js

```diff
@@ -0,0 +1,162 @@
+const fse = require('fs-extra'),
+    Logger = require('../logger.js'),
+    logger = new Logger('netsblox:jsdoc'),
+    doctrine = require('doctrine');
+
+
+const MARKER_START = '/**',
+    MARKER_START_SKIP = '/***',
+    MARKER_END = '*/';
+
+let parseSync = (filePath, searchScope = 5) => {
+    let source = fse.readFileSync(filePath, 'UTF8');
+    return parseSource(source, searchScope);
+};
+
+function parseSource(source, searchScope){
+    let lines = source.split(/\n/);
+    let blocks = extractDocBlocks(source);
+    blocks = blocks.map(block => {
+        let src = block.lines.join('\n');
+        block.parsed = doctrine.parse(src, {unwrap: true});
+        return block;
+    });
+
+    blocks = blocks.filter(block => {
+        let linesToSearch = lines.slice(block.endLine, block.endLine + searchScope);
+        let fnName;
+        // if @name is set jsut use that and save a few cycles
+        if (block.parsed.tags.some(tag => tag.title === 'name')) {
+            fnName  = block.parsed.tags.find(tag => tag.title === 'name').name;
+            logger.info('fn name set through @name', fnName);
+        } else {
+            fnName = findFn(linesToSearch);
+            if (!fnName){
+                logger.error(`can't associate ${block.lines} with any function. # Fix it at line ${block.beginLine}, column ${block.column}`);
```

Comment:
I would change this to a warning instead of an error. Also, I would print the actual comment that you cannot parse instead of just the line row and column.

Reply:
.lines is the actual lines